awebox/ocp/constraints.py

```diff
@@ -145,17 +116,60 @@ def get_constraints(nlp_options, V, P, Xdot, model, dae, formulation, Integral_c
 
     return ocp_cstr_list, ocp_cstr_struct
 
+def get_subset_of_shooting_node_equalities_that_wont_cause_licq_errors(model):
+
+    model_constraints_list = model.constraints_list
+    model_variables = model.variables
+
+    # remove those constraints that only depend on 'xd', since continuity will duplicate those...
+    relevant_shooting_vars = []
+    for var_type in (set(model_variables.keys()) - set(['xd'])):
+        relevant_shooting_vars = cas.vertcat(relevant_shooting_vars, model_variables[var_type])
+    mdl_shooting_cstr_sublist = mdl_constraint.MdlConstraintList()
+
+    for cstr in model_constraints_list.get_list('eq'):
+
+        cstr_expr = cstr.expr
+        selected_expr = []
+
+        for cdx in range(cstr_expr.shape[0]):
+            local_expr = cstr_expr[cdx]
+            local_jac = cas.jacobian(local_expr, relevant_shooting_vars)
+
+            will_be_dependent = not (local_jac.nnz() > 0)
+            if will_be_dependent:
+                message = 'the ' + str(cdx) + 'th entry of the model ' + cstr.name + ' equality constraint ' \
+                        'would be likely to trigger licq violations in direct collocation, so the conflicting '\
+                        'constraints have not been enforced at the shooting nodes. we suggest re-considering '\
+                        'this constraint formulation.'
+                awelogger.logger.warning(message)
+            else:
+                selected_expr = cas.vertcat(selected_expr, local_expr)
+
+        # if selected_expr is not still []
+        if not isinstance(selected_expr, list):
+            selected_cstr = cstr_op.Constraint(expr=selected_expr,
+                                               name=cstr.name + '_selected',
+                                               cstr_type='eq')
+            mdl_shooting_cstr_sublist.append(selected_cstr)
+
+    return mdl_shooting_cstr_sublist
 
```

Comment:
This seems useful! Could you shortly comment on the typical use cases when this behavior would be triggered?

Reply:
sure. i had a (badly thought through, has since been removed, etc) constraint that depended only on state variables. since I didn't realize that this was triggering an licq problem, i had to hunt a lot... so, it would be nice if this sort of bug were automatically detected and the user warned, but so that the problem still solves...